addons/docs/src/lib/convert/convert.test.ts

```diff
@@ -603,9 +604,17 @@ describe('storybook type system', () => {
           "oneOfNumber": {
             "name": "enum",
             "value": [
-              "1",
-              "2",
-              "3"
+              1,
+              2,
+              3
+            ]
+          },
+          "oneOfStringNumber": {
+            "name": "enum",
+            "value": [
+              1,
```

Comment:
Hey @yahma25 actually this should be of a string value. Can you check this? Thanks!

Reply:
Okay @yannbf 

When I printed out the console:
```ts
console.log(convertJs(input));
```

The result is:
```bash
 console.log
    {
      oneOfNumber: { name: 'enum', value: [ 1, 2, 3 ] }, # <-- number
      oneOfStringNumber: { name: 'enum', value: [ 1, 2, 3 ] }, # <-- number
      oneOfString: { name: 'enum', value: [ 'static', 'timed' ] }
    }
```
because of the [`addons/docs/src/lib/convert/proptypes/convert.ts` I committed](https://github.com/storybookjs/storybook/pull/15822/files/3a3561e3c68c65ad4e3f0fe320ceff2230e860db#diff-13498078db56d168675611075c92f5ee4ea0437351f26236889ad48b2a684410)
* addons/docs/src/lib/convert/proptypes/convert.ts
```ts
// ...
case 'enum': {
      const values = computed ? value : value.map((v: PTType) => trimQuotes(v.value));
      const values = computed
        ? value
        : value.map((v: PTType) => {
            const trimmedValue = trimQuotes(v.value);
            return Number.isNaN(Number(trimmedValue)) ? trimmedValue : Number(trimmedValue);
          });
      return { ...base, name, value: values };
    }
// ...
```

Did I misunderstand? 🙄

Reply:
I think it could be related to the assertion happening after `trimQuotes` is used, which makes isNaN return false in both cases.

If your enum is of numbers, the values will be like `"2"`, in comparison to when the enum is of strings that also happen to be a number, and the values will be like `"'2'"`. 

I might be wrong but I believe in this case users will want the values as string because that's what the component expects, and might break their code if they manipulate the value as a string but receive a number instead.

So I believe the expected result should be:

```js
 console.log(
    {
      oneOfNumber: { name: 'enum', value: [ 1, 2, 3 ] }, # <-- number
      oneOfStringNumber: { name: 'enum', value: [ '1', '2', '3' ] }, # <-- string
      oneOfString: { name: 'enum', value: [ 'static', 'timed' ] }
    }
)
```

@ndelangen @shilman

Reply:
@yannbf 
I understand your meaning and thank you for explaining. 😄 
I agree with your answer. 
please check my up-to-date code. 🙏